luamocks/autogenerated/SyncedCtrl.lua

```diff
@@ -0,0 +1,1486 @@
+
+---Parameters
+---@param featureID number
+---@param rotX number
+---@param rotY number
+---@param rotZ number
+---@return nil
+function Spring.SetFeatureRotation(featureID, rotX, rotY, rotZ) end
+
+---Parameters
+---@param featureID number
+---@param dirX number
+---@param dirY number
+---@param dirZ number
+---@return nil
+function Spring.SetFeatureDirection(featureID, dirX, dirY, dirZ) end
+
+---Parameters
+---@param featureID number
+---@param heading number
+---@param upx number
+---@param upy number
+---@param upz number
+---@return nil
+function Spring.SetFeatureHeadingAndUpDir(featureID, heading, upx, upy, upz) end
+
+---Parameters
+---@param featureID number
+---@param velX number
+---@param velY number
+---@param velZ number
+---@return nil
+function Spring.SetFeatureVelocity(featureID, velX, velY, velZ) end
+
+---Parameters
+---@param featureID number
+---@param isBlocking boolean
+---@param isSolidObjectCollidable boolean
+---@param isProjectileCollidable boolean
+---@param isRaySegmentCollidable boolean
+---@param crushable boolean
+---@param blockEnemyPushing boolean
+---@param blockHeightChanges boolean
+---@return nil
+function Spring.SetFeatureBlocking(featureID, isBlocking, isSolidObjectCollidable, isProjectileCollidable, isRaySegmentCollidable, crushable, blockEnemyPushing, blockHeightChanges) end
+
+---Parameters
+---@param featureID number
+---@param noSelect boolean
+---@return nil
+function Spring.SetFeatureNoSelect(featureID, noSelect) end
+
+---Parameters
+---@param featureID number
+---@param mpX number
+---@param mpY number
+---@param mpZ number
+---@param apX number
+---@param apY number
+---@param apZ number
+---@param relative boolean (optional)
+---@return bool success
+function Spring.SetFeatureMidAndAimPos(featureID, mpX, mpY, mpZ, apX, apY, apZ[, relative]) end
+
+---Parameters
+---@param featureID number
+---@param radius number
+---@param height number
+---@return bool success
+function Spring.SetFeatureRadiusAndHeight(featureID, radius, height) end
+
+---Parameters
+---@param featureID number
+---@param scaleX number
+---@param scaleY number
+---@param scaleZ number
+---@param offsetX number
+---@param offsetY number
+---@param offsetZ number
+---@param vType number
+---@param tType number
+---@param Axis number
+---@return nil
+function Spring.SetFeatureCollisionVolumeData(featureID, scaleX, scaleY, scaleZ, offsetX, offsetY, offsetZ, vType, tType, Axis) end
+
+---Parameters
+---@param featureID number
+---@param pieceIndex number
+---@param enable boolean
+---@param scaleX number
+---@param scaleY number
+---@param scaleZ number
+---@param offsetX number
+---@param offsetY number
+---@param offsetZ number
+---@param Axis number
+---@param volumeType number
+---@param primaryAxis number (optional)
+---@return nil
+function Spring.SetFeaturePieceCollisionVolumeData(featureID, pieceIndex, enable, scaleX, scaleY, scaleZ, offsetX, offsetY, offsetZ, Axis, volumeType[, primaryAxis]) end
```

Comment:
the `[, primaryAxis]` doesn't look like valid lua, will that be a problem?

Reply:
This seems to be fine for LuaLS to parse. I do not know if it will present an issue otherwise, but eventually it will be stripped ideally in the script that generates these files.

Reply:
![image](https://github.com/beyond-all-reason/Beyond-All-Reason/assets/806377/b9878ac7-dcea-42d1-a464-10f01cc22b39)
This is what it produces as configured on my copy of VSCode.